conf/reflect.go

```diff
@@ -0,0 +1,76 @@
+package conf
+
+import (
+	"errors"
+	"fmt"
+	"reflect"
+
+	"github.com/spf13/viper"
+)
+
+const tagPrefix = "viper"
+
+func populateConfig(config *Configuration) (*Configuration, error) {
+	err := recursivelySet(reflect.ValueOf(config), "")
+	if err != nil {
+		return nil, err
+	}
+
+	return config, nil
+}
+
+func recursivelySet(val reflect.Value, prefix string) error {
+	if val.Kind() != reflect.Ptr {
+		return errors.New("WTF")
+	}
+
+	// dereference
+	val = reflect.Indirect(val)
+	if val.Kind() != reflect.Struct {
+		return errors.New("FML")
+	}
+
+	// grab the type for this instance
+	vType := reflect.TypeOf(val.Interface())
+
+	// go through child fields
+	for i := 0; i < val.NumField(); i++ {
+		thisField := val.Field(i)
+		thisType := vType.Field(i)
+		tag := prefix + getTag(thisType)
+
+		switch thisField.Kind() {
```

Comment:
This is missing boolean values. See https://github.com/netlify/netlify-auth/blob/master/conf/reflect.go#L81